src/TabViewPanResponder.js

```diff
@@ -55,11 +55,14 @@ function forHorizontal(props: Props) {
 
   function canMoveScreen(evt: GestureEvent, gestureState: GestureState) {
     const { routes, index } = props.navigationState;
-    return (
-      isMovingHorzontally(evt, gestureState) && (
+    const canMove = isMovingHorzontally(evt, gestureState) && (
         (gestureState.dx >= 0 && index >= 0) ||
         (gestureState.dx <= -0 && index <= routes.length - 1)
-    ));
+      );
+    if (canMove) {
+      startDirection = gestureState.dx;
+    }
+    return canMove;
   }
```

Comment:
Moved the startingDirection capture to this function instead of making a separate one.